{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "fixed-portland",
   "metadata": {},
   "outputs": [],
   "source": [
    "library('SQMtools')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "middle-locator",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "   user  system elapsed \n",
       "253.711  10.595 264.367 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "system.time(load(file=\"./RDatas/oil_spades.RData\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "floral-coordinator",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "'oil_spades'"
      ],
      "text/latex": [
       "'oil\\_spades'"
      ],
      "text/markdown": [
       "'oil_spades'"
      ],
      "text/plain": [
       "[1] \"oil_spades\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "ls()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "floppy-nature",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "List of 7\n",
      " $ misc       :List of 11\n",
      "  ..$ project_name     : chr \"HN00149294_RawData_squeezemeta_spades_0623\"\n",
      "  ..$ samples          : chr [1:3] \"ECHO_Blank\" \"ECHO_C\" \"ECHO_F\"\n",
      "  ..$ tax_names_long   :List of 7\n",
      "  .. ..$ superkingdom: Named chr [1:7] \"k_Archaea\" \"k_Bacteria\" \"k_Eukaryota\" \"k_No CDS\" ...\n",
      "  .. .. ..- attr(*, \"names\")= chr [1:7] \"Archaea\" \"Bacteria\" \"Eukaryota\" \"No CDS\" ...\n",
      "  .. ..$ phylum      : Named chr [1:242] \"k_Archaea;p_Candidatus Aenigmarchaeota\" \"k_Archaea;p_Candidatus Altiarchaeota\" \"k_Archaea;p_Candidatus Bathyarchaeota\" \"k_Archaea;p_Candidatus Diapherotrites\" ...\n",
      "  .. .. ..- attr(*, \"names\")= chr [1:242] \"Candidatus Aenigmarchaeota\" \"Candidatus Altiarchaeota\" \"Candidatus Bathyarchaeota\" \"Candidatus Diapherotrites\" ...\n",
      "  .. ..$ class       : Named chr [1:542] \"k_Archaea;p_Candidatus Aenigmarchaeota;c_Unclassified Candidatus Aenigmarchaeota\" \"k_Archaea;p_Candidatus Altiarchaeota;c_Candidatus Altiarchaeales (no class in NCBI)\" \"k_Archaea;p_Candidatus Altiarchaeota;c_Unclassified Candidatus Altiarchaeota\" \"k_Archaea;p_Candidatus Bathyarchaeota;c_Candidatus Bathyarchaeota archaeon (no class in NCBI)\" ...\n",
      "  .. .. ..- attr(*, \"names\")= chr [1:542] \"Unclassified Candidatus Aenigmarchaeota\" \"Candidatus Altiarchaeales (no class in NCBI)\" \"Unclassified Candidatus Altiarchaeota\" \"Candidatus Bathyarchaeota archaeon (no class in NCBI)\" ...\n",
      "  .. ..$ order       : Named chr [1:978] \"k_Archaea;p_Candidatus Aenigmarchaeota;c_Unclassified Candidatus Aenigmarchaeota;o_Unclassified Candidatus Aenigmarchaeota\" \"k_Archaea;p_Candidatus Altiarchaeota;c_Candidatus Altiarchaeales (no class in NCBI);o_Candidatus Altiarchaeales\" \"k_Archaea;p_Candidatus Altiarchaeota;c_Unclassified Candidatus Altiarchaeota;o_Unclassified Candidatus Altiarchaeota\" \"k_Archaea;p_Candidatus Bathyarchaeota;c_Candidatus Bathyarchaeota archaeon (no class in NCBI);o_Candidatus Bath\"| __truncated__ ...\n",
      "  .. .. ..- attr(*, \"names\")= chr [1:978] \"Unclassified Candidatus Aenigmarchaeota\" \"Candidatus Altiarchaeales\" \"Unclassified Candidatus Altiarchaeota\" \"Candidatus Bathyarchaeota archaeon (no order in NCBI)\" ...\n",
      "  .. ..$ family      : Named chr [1:1751] \"k_Archaea;p_Candidatus Aenigmarchaeota;c_Unclassified Candidatus Aenigmarchaeota;o_Unclassified Candidatus Aeni\"| __truncated__ \"k_Archaea;p_Candidatus Altiarchaeota;c_Candidatus Altiarchaeales (no class in NCBI);o_Candidatus Altiarchaeales\"| __truncated__ \"k_Archaea;p_Candidatus Altiarchaeota;c_Candidatus Altiarchaeales (no class in NCBI);o_Candidatus Altiarchaeales\"| __truncated__ \"k_Archaea;p_Candidatus Altiarchaeota;c_Unclassified Candidatus Altiarchaeota;o_Unclassified Candidatus Altiarch\"| __truncated__ ...\n",
      "  .. .. ..- attr(*, \"names\")= chr [1:1751] \"Unclassified Candidatus Aenigmarchaeota\" \"Candidatus Altiarchaeum (no family in NCBI)\" \"Unclassified Candidatus Altiarchaeales\" \"Unclassified Candidatus Altiarchaeota\" ...\n",
      "  .. ..$ genus       : Named chr [1:3688] \"k_Archaea;p_Candidatus Aenigmarchaeota;c_Unclassified Candidatus Aenigmarchaeota;o_Unclassified Candidatus Aeni\"| __truncated__ \"k_Archaea;p_Candidatus Altiarchaeota;c_Candidatus Altiarchaeales (no class in NCBI);o_Candidatus Altiarchaeales\"| __truncated__ \"k_Archaea;p_Candidatus Altiarchaeota;c_Candidatus Altiarchaeales (no class in NCBI);o_Candidatus Altiarchaeales\"| __truncated__ \"k_Archaea;p_Candidatus Altiarchaeota;c_Unclassified Candidatus Altiarchaeota;o_Unclassified Candidatus Altiarch\"| __truncated__ ...\n",
      "  .. .. ..- attr(*, \"names\")= chr [1:3688] \"Unclassified Candidatus Aenigmarchaeota\" \"Candidatus Altiarchaeum\" \"Unclassified Candidatus Altiarchaeales\" \"Unclassified Candidatus Altiarchaeota\" ...\n",
      "  .. ..$ species     : Named chr [1:6252] \"k_Archaea;p_Candidatus Aenigmarchaeota;c_Unclassified Candidatus Aenigmarchaeota;o_Unclassified Candidatus Aeni\"| __truncated__ \"k_Archaea;p_Candidatus Altiarchaeota;c_Candidatus Altiarchaeales (no class in NCBI);o_Candidatus Altiarchaeales\"| __truncated__ \"k_Archaea;p_Candidatus Altiarchaeota;c_Candidatus Altiarchaeales (no class in NCBI);o_Candidatus Altiarchaeales\"| __truncated__ \"k_Archaea;p_Candidatus Altiarchaeota;c_Unclassified Candidatus Altiarchaeota;o_Unclassified Candidatus Altiarch\"| __truncated__ ...\n",
      "  .. .. ..- attr(*, \"names\")= chr [1:6252] \"Unclassified Candidatus Aenigmarchaeota\" \"Candidatus Altiarchaeum sp.\" \"Unclassified Candidatus Altiarchaeales\" \"Unclassified Candidatus Altiarchaeota\" ...\n",
      "  ..$ tax_names_short  : Named chr [1:13460] \"Archaea\" \"Bacteria\" \"Eukaryota\" \"No CDS\" ...\n",
      "  .. ..- attr(*, \"names\")= chr [1:13460] \"k_Archaea\" \"k_Bacteria\" \"k_Eukaryota\" \"k_No CDS\" ...\n",
      "  ..$ ext_annot_sources: list()\n",
      "  ..$ KEGG_names       : Named chr [1:12478] \"alcohol dehydrogenase [EC:1.1.1.1]\" \"alcohol dehydrogenase (NADP+) [EC:1.1.1.2]\" \"homoserine dehydrogenase [EC:1.1.1.3]\" \"(R,R)-butanediol dehydrogenase / meso-butanediol dehydrogenase / diacetyl reductase [EC:1.1.1.4 1.1.1.- 1.1.1.303]\" ...\n",
      "  .. ..- attr(*, \"names\")= chr [1:12478] \"K00001\" \"K00002\" \"K00003\" \"K00004\" ...\n",
      "  ..$ KEGG_paths       : Named chr [1:12478] \"Metabolism; Carbohydrate metabolism; Glycolysis / Gluconeogenesis | Metabolism; Lipid metabolism; Fatty acid de\"| __truncated__ \"Metabolism; Carbohydrate metabolism; Glycolysis / Gluconeogenesis | Metabolism; Carbohydrate metabolism; Pentos\"| __truncated__ \"Metabolism; Amino acid metabolism; Glycine, serine and threonine metabolism | Metabolism; Amino acid metabolism\"| __truncated__ \"Metabolism; Carbohydrate metabolism; Butanoate metabolism\" ...\n",
      "  .. ..- attr(*, \"names\")= chr [1:12478] \"K00001\" \"K00002\" \"K00003\" \"K00004\" ...\n",
      "  ..$ COG_names        : Named chr [1:55335] \"Glutamate-1-semialdehyde aminotransferase\" \"Acetylglutamate semialdehyde dehydrogenase\" \"Oxyanion-translocating ATPase\" \"Ammonia permease\" ...\n",
      "  .. ..- attr(*, \"names\")= chr [1:55335] \"COG0001\" \"COG0002\" \"COG0003\" \"COG0004\" ...\n",
      "  ..$ COG_paths        : Named chr [1:55335] \"Coenzyme metabolism\" \"Amino acid transport and metabolism\" \"Inorganic ion transport and metabolism\" \"Inorganic ion transport and metabolism\" ...\n",
      "  .. ..- attr(*, \"names\")= chr [1:55335] \"COG0001\" \"COG0002\" \"COG0003\" \"COG0004\" ...\n",
      "  ..$ coding_fraction  :List of 3\n",
      "  .. ..$ KEGG: Named num [1:3] 0.551 0.543 0.473\n",
      "  .. .. ..- attr(*, \"names\")= chr [1:3] \"ECHO_Blank\" \"ECHO_C\" \"ECHO_F\"\n",
      "  .. ..$ COG : Named num [1:3] 0.551 0.543 0.473\n",
      "  .. .. ..- attr(*, \"names\")= chr [1:3] \"ECHO_Blank\" \"ECHO_C\" \"ECHO_F\"\n",
      "  .. ..$ PFAM: Named num [1:3] 0.542 0.535 0.465\n",
      "  .. .. ..- attr(*, \"names\")= chr [1:3] \"ECHO_Blank\" \"ECHO_C\" \"ECHO_F\"\n",
      "  ..$ RecA_cov         : Named num [1:3] 3231 3072 3210\n",
      "  .. ..- attr(*, \"names\")= chr [1:3] \"ECHO_Blank\" \"ECHO_C\" \"ECHO_F\"\n",
      " $ orfs       :List of 7\n",
      "  ..$ table:Classes ‘generic.data.frame’ and 'data.frame':\t8410901 obs. of  28 variables:\n",
      "  .. ..$ Contig ID                : chr [1:8410901] \"megahit_1\" \"megahit_1\" \"megahit_1\" \"megahit_1\" ...\n",
      "  .. ..$ Molecule                 : chr [1:8410901] \"CDS\" \"CDS\" \"CDS\" \"tRNA\" ...\n",
      "  .. ..$ Method                   : chr [1:8410901] \"Prodigal\" \"Prodigal\" \"Prodigal\" \"Aragorn\" ...\n",
      "  .. ..$ Length NT                : int [1:8410901] 324 621 639 91 747 984 771 567 960 171 ...\n",
      "  .. ..$ Length AA                : int [1:8410901] 108 207 213 NA 249 328 257 189 320 57 ...\n",
      "  .. ..$ GC perc                  : num [1:8410901] 84 74.2 71.2 0 75 ...\n",
      "  .. ..$ Gene name                : chr [1:8410901] \"rpoE\" \"ogt, MGMT\" \"\" \"tRNA-Ser\" ...\n",
      "  .. ..$ Tax                      : chr [1:8410901] \"k_Bacteria;n_Terrabacteria group;p_Actinobacteria;c_Thermoleophilia\" \"k_Bacteria;n_Terrabacteria group;p_Actinobacteria;c_Thermoleophilia;o_Thermoleophilales;f_Thermoleophilaceae\" \"k_Bacteria;n_Terrabacteria group;p_Actinobacteria;c_Thermoleophilia;o_Solirubrobacterales\" \"\" ...\n",
      "  .. ..$ KEGG ID                  : chr [1:8410901] \"K03088*\" \"K00567*\" \"\" \"\" ...\n",
      "  .. ..$ KEGGFUN                  : chr [1:8410901] \"RNA polymerase sigma-70 factor, ECF subfamily\" \"methylated-DNA-[protein]-cysteine S-methyltransferase [EC:2.1.1.63]\" \"\" \"\" ...\n",
      "  .. ..$ KEGGPATH                 : chr [1:8410901] \"Brite Hierarchies; Protein families: genetic information processing; Transcription machinery\" \"Brite Hierarchies; Protein families: genetic information processing; DNA repair and recombination proteins\" \"\" \"\" ...\n",
      "  .. ..$ COG ID                   : chr [1:8410901] \"ENOG4111P7H\" \"COG0350*\" \"\" \"\" ...\n",
      "  .. ..$ COGFUN                   : chr [1:8410901] \"R Polymerase\" \"Methylated DNA-protein cysteine methyltransferase\" \"\" \"\" ...\n",
      "  .. ..$ COGPATH                  : chr [1:8410901] \"Transcription\" \"DNA replication, recombination and repair\" \"\" \"\" ...\n",
      "  .. ..$ PFAM                     : chr [1:8410901] \"\" \"PF01035 [6-O-methylguanine DNA methyltransferase, DNA binding domain]\" \"\" \"\" ...\n",
      "  .. ..$ TPM ECHO_Blank           : num [1:8410901] 1.854 0.563 0.649 1.68 0.906 ...\n",
      "  .. ..$ TPM ECHO_C               : num [1:8410901] 1.893 0.659 0.488 0.946 0.533 ...\n",
      "  .. ..$ TPM ECHO_F               : num [1:8410901] 1.528 0.418 0.387 0.259 0.426 ...\n",
      "  .. ..$ Coverage ECHO_Blank      : num [1:8410901] 23.23 5.16 7.58 8.66 9.93 ...\n",
      "  .. ..$ Coverage ECHO_C          : num [1:8410901] 22.53 8.03 6.3 5.11 6.24 ...\n",
      "  .. ..$ Coverage ECHO_F          : num [1:8410901] 18.731 4.224 3.917 0.396 4.118 ...\n",
      "  .. ..$ Raw read count ECHO_Blank: int [1:8410901] 55 32 38 14 62 92 77 53 93 26 ...\n",
      "  .. ..$ Raw read count ECHO_C    : int [1:8410901] 57 38 29 8 37 76 85 49 70 19 ...\n",
      "  .. ..$ Raw read count ECHO_F    : int [1:8410901] 42 22 21 2 27 74 38 25 40 14 ...\n",
      "  .. ..$ Raw base count ECHO_Blank: int [1:8410901] 7525 3204 4841 788 7416 11667 9996 5980 11886 1954 ...\n",
      "  .. ..$ Raw base count ECHO_C    : int [1:8410901] 7301 4985 4025 465 4662 9926 11325 5903 9290 1610 ...\n",
      "  .. ..$ Raw base count ECHO_F    : int [1:8410901] 6069 2623 2503 36 3076 9570 5124 3424 5186 1289 ...\n",
      "  .. ..$ Hits                     : int [1:8410901] 25 25 1 0 25 25 25 25 25 1 ...\n",
      "  ..$ abund: num [1:8410901, 1:3] 55 32 38 14 62 92 77 53 93 26 ...\n",
      "  .. ..- attr(*, \"dimnames\")=List of 2\n",
      "  .. .. ..$ : chr [1:8410901] \"megahit_1_2-325\" \"megahit_1_322-942\" \"megahit_1_944-1582\" \"megahit_1_1650-1740\" ...\n",
      "  .. .. ..$ : chr [1:3] \"ECHO_Blank\" \"ECHO_C\" \"ECHO_F\"\n",
      "  ..$ bases: num [1:8410901, 1:3] 7525 3204 4841 788 7416 ...\n",
      "  .. ..- attr(*, \"dimnames\")=List of 2\n",
      "  .. .. ..$ : chr [1:8410901] \"megahit_1_2-325\" \"megahit_1_322-942\" \"megahit_1_944-1582\" \"megahit_1_1650-1740\" ...\n",
      "  .. .. ..$ : chr [1:3] \"ECHO_Blank\" \"ECHO_C\" \"ECHO_F\"\n",
      "  ..$ cov  : num [1:8410901, 1:3] 23.23 5.16 7.58 8.66 9.93 ...\n",
      "  .. ..- attr(*, \"dimnames\")=List of 2\n",
      "  .. .. ..$ : chr [1:8410901] \"megahit_1_2-325\" \"megahit_1_322-942\" \"megahit_1_944-1582\" \"megahit_1_1650-1740\" ...\n",
      "  .. .. ..$ : chr [1:3] \"ECHO_Blank\" \"ECHO_C\" \"ECHO_F\"\n",
      "  ..$ tpm  : num [1:8410901, 1:3] 1.854 0.563 0.649 1.68 0.906 ...\n",
      "  .. ..- attr(*, \"dimnames\")=List of 2\n",
      "  .. .. ..$ : chr [1:8410901] \"megahit_1_2-325\" \"megahit_1_322-942\" \"megahit_1_944-1582\" \"megahit_1_1650-1740\" ...\n",
      "  .. .. ..$ : chr [1:3] \"ECHO_Blank\" \"ECHO_C\" \"ECHO_F\"\n",
      "  ..$ seqs : Named chr [1:8410901] \"PPPPPPPPPPPPPPPPPRPPARPPRPPPPPPPPPPPPPPPPPPPPPPPPRLWSRVRALPEKQRMAVFLRSVADLDYAEVAAALDCSQDAARRNVHEGLKRLREELAR*\" \"MNEVDRQLRDAVRESRHGGPSQRALDELAARAAREGLVDVAYGQADSPFGPLTVAATDRGLVLLAYPELAVDDVLDRLAREVSPRVLELPARIEPIRRQLDEYFEGRRRAF\"| __truncated__ \"MAAMKIFVAVVIGVVVLGAVTQLVIPGFVERKIESNLKERSDAGEVVADVKAFPAVRLLWGSGDRLEVRGRGLKIDLAERTDDPLGQIDGFDEVDIDLTDLTAGPVDVQAF\"| __truncated__ \"CGGAGAGGTGGCCGAGTGGCTGAAGGCACTCGCCTGCTAAGCGAGTAGGGGGCGAAAGCTCTCTCGCGGGTTCGAATCCCGCCCTCTCCGT\" ...\n",
      "  .. ..- attr(*, \"names\")= chr [1:8410901] \"megahit_1_2-325\" \"megahit_1_322-942\" \"megahit_1_944-1582\" \"megahit_1_1650-1740\" ...\n",
      "  ..$ tax  : chr [1:8410901, 1:7] \"Bacteria\" \"Bacteria\" \"Bacteria\" \"Bacteria\" ...\n",
      "  .. ..- attr(*, \"dimnames\")=List of 2\n",
      "  .. .. ..$ : chr [1:8410901] \"megahit_1000000_2-667\" \"megahit_1000001_2-667\" \"megahit_1000002_19-669\" \"megahit_1000003_128-667\" ...\n",
      "  .. .. ..$ : chr [1:7] \"superkingdom\" \"phylum\" \"class\" \"order\" ...\n",
      " $ contigs    :List of 7\n",
      "  ..$ table:Classes ‘generic.data.frame’ and 'data.frame':\t6219967 obs. of  15 variables:\n",
      "  .. ..$ Tax                      : chr [1:6219967] \"k_Bacteria;n_Terrabacteria group;p_Actinobacteria;c_Thermoleophilia\" \"k_Bacteria;n_Terrabacteria group;p_Actinobacteria;c_Actinobacteria\" \"k_Bacteria;n_Terrabacteria group;p_Actinobacteria;c_Thermoleophilia\" \"k_Bacteria;n_FCB group;p_Gemmatimonadetes\" ...\n",
      "  .. ..$ Disparity                : num [1:6219967] 0.407 0.036 0.359 0.193 0.065 0.052 0.382 0.06 0.024 0.062 ...\n",
      "  .. ..$ GC perc                  : num [1:6219967] 71 70.9 70.7 70.8 72.7 ...\n",
      "  .. ..$ Length                   : int [1:6219967] 812687 783958 683686 531294 440409 414181 372763 348188 302316 293087 ...\n",
      "  .. ..$ Num genes                : int [1:6219967] 784 720 694 445 386 390 351 289 287 238 ...\n",
      "  .. ..$ Bin ID                   : chr [1:6219967] \"\" \"\" \"\" \"\" ...\n",
      "  .. ..$ Coverage ECHO_Blank      : num [1:6219967] 10.77 10.54 11.01 4.63 3.62 ...\n",
      "  .. ..$ TPM ECHO_Blank           : num [1:6219967] 1.4 1.3 1.4 0.6 0.5 0.5 1.4 0.5 0.5 0.5 ...\n",
      "  .. ..$ Raw read count ECHO_Blank: int [1:6219967] 57990 54727 49829 16294 10561 11070 26834 8377 7366 7911 ...\n",
      "  .. ..$ Coverage ECHO_C          : num [1:6219967] 9.75 9.56 9.86 3.41 29.14 ...\n",
      "  .. ..$ TPM ECHO_C               : num [1:6219967] 1.2 1.2 1.2 0.4 3.5 0.4 1.2 3.6 3.6 0.4 ...\n",
      "  .. ..$ Raw read count ECHO_C    : int [1:6219967] 52456 49634 44644 12007 84995 8014 24677 67648 58916 5957 ...\n",
      "  .. ..$ Coverage ECHO_F          : num [1:6219967] 6.71 6.59 6.94 17.59 5.69 ...\n",
      "  .. ..$ TPM ECHO_F               : num [1:6219967] 0.8 0.8 0.9 2.2 0.7 2 0.8 0.7 0.7 2.1 ...\n",
      "  .. ..$ Raw read count ECHO_F    : int [1:6219967] 36137 34203 31433 61893 16586 43862 16373 13450 11767 31559 ...\n",
      "  ..$ abund: num [1:6219967, 1:3] 57990 54727 49829 16294 10561 ...\n",
      "  .. ..- attr(*, \"dimnames\")=List of 2\n",
      "  .. .. ..$ : chr [1:6219967] \"megahit_1\" \"megahit_2\" \"megahit_3\" \"megahit_4\" ...\n",
      "  .. .. ..$ : chr [1:3] \"ECHO_Blank\" \"ECHO_C\" \"ECHO_F\"\n",
      "  ..$ cov  : num [1:6219967, 1:3] 10.77 10.54 11.01 4.63 3.62 ...\n",
      "  .. ..- attr(*, \"dimnames\")=List of 2\n",
      "  .. .. ..$ : chr [1:6219967] \"megahit_1\" \"megahit_2\" \"megahit_3\" \"megahit_4\" ...\n",
      "  .. .. ..$ : chr [1:3] \"ECHO_Blank\" \"ECHO_C\" \"ECHO_F\"\n",
      "  ..$ tpm  : num [1:6219967, 1:3] 1.4 1.3 1.4 0.6 0.5 0.5 1.4 0.5 0.5 0.5 ...\n",
      "  .. ..- attr(*, \"dimnames\")=List of 2\n",
      "  .. .. ..$ : chr [1:6219967] \"megahit_1\" \"megahit_2\" \"megahit_3\" \"megahit_4\" ...\n",
      "  .. .. ..$ : chr [1:3] \"ECHO_Blank\" \"ECHO_C\" \"ECHO_F\"\n",
      "  ..$ seqs : Named chr [1:6219967] \"CCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCGCCCCCCCGCCCGCCCCCCCCGCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCC\"| __truncated__ \"CGGAGCGCTGAAGGACGATGGGGCTATCGTCCAAGAGGGCGAGAAGCGGGGTGCGAAGTACACGCCTGCTTCGTAGGAGGGAATGATAAGGTGGTGTACTGGTCTTCGGGT\"| __truncated__ \"AACAACCACCGGGTAAGTTGCTAAAGGATTTCCTTTAGCGTCTGTTTTAGGCGCGGGGATTAAACCGCTAGCTTTCTCTTCATTAGAAAGATATTCTGTTTCCGGCTTCGG\"| __truncated__ \"GCCTACTTCACGCACGTCACCTCGCTCGTTCACCCCACGCCGCAGCCCCTTCCTAAGGCTGCATAACCCGCCCGCCGATCCACTTATCTCAGGCCTCAAACCTGTTCAGAA\"| __truncated__ ...\n",
      "  .. ..- attr(*, \"names\")= chr [1:6219967] \"megahit_1\" \"megahit_2\" \"megahit_3\" \"megahit_4\" ...\n",
      "  ..$ tax  : chr [1:6219967, 1:7] \"Bacteria\" \"Bacteria\" \"Bacteria\" \"Bacteria\" ...\n",
      "  .. ..- attr(*, \"dimnames\")=List of 2\n",
      "  .. .. ..$ : chr [1:6219967] \"megahit_1\" \"megahit_2\" \"megahit_3\" \"megahit_4\" ...\n",
      "  .. .. ..$ : chr [1:7] \"superkingdom\" \"phylum\" \"class\" \"order\" ...\n",
      "  ..$ bins : chr [1:6219967, 1] \"No_bin\" \"No_bin\" \"No_bin\" \"No_bin\" ...\n",
      "  .. ..- attr(*, \"dimnames\")=List of 2\n",
      "  .. .. ..$ : chr [1:6219967] \"megahit_1\" \"megahit_2\" \"megahit_3\" \"megahit_4\" ...\n",
      "  .. .. ..$ : chr \"DASTool\"\n",
      " $ bins       :List of 3\n",
      "  ..$ table:'data.frame':\t82 obs. of  16 variables:\n",
      "  .. ..$ Method             : chr [1:82] \"DASTool\" \"DASTool\" \"DASTool\" \"DASTool\" ...\n",
      "  .. ..$ Tax                : chr [1:82] \"k_Bacteria;n_FCB group;n_Bacteroidetes/Chlorobi group;p_Bacteroidetes\" \"k_Bacteria;n_Terrabacteria group;p_Actinobacteria;c_Actinobacteria;o_Streptomycetales;f_Streptomycetaceae;g_Streptomyces\" \"k_Bacteria;p_Proteobacteria;c_Gammaproteobacteria\" \"k_Bacteria;n_FCB group;n_Bacteroidetes/Chlorobi group;p_Bacteroidetes\" ...\n",
      "  .. ..$ Tax 16S            : chr [1:82] \"superkingdom:Bacteria;clade:FCB group;clade:Bacteroidetes/Chlorobi group;phylum:Bacteroidetes;class:Chitinophag\"| __truncated__ \"\" \"superkingdom:Bacteria;phylum:Proteobacteria;class:Gammaproteobacteria\" \"\" ...\n",
      "  .. ..$ Length             : int [1:82] 4354063 6285984 4863781 4068638 5709043 4296532 5038503 3726492 3461875 5150574 ...\n",
      "  .. ..$ GC perc            : num [1:82] 43 73.2 66.6 48.4 34 ...\n",
      "  .. ..$ Num contigs        : int [1:82] 284 66 1128 164 2477 1323 1049 478 327 1670 ...\n",
      "  .. ..$ Disparity          : num [1:82] 0 0 0.539 0 0.668 0.191 0.376 0.018 0 0.373 ...\n",
      "  .. ..$ Completeness       : num [1:82] 95.1 94.5 93.5 93 92.8 ...\n",
      "  .. ..$ Contamination      : num [1:82] 1.03 1.61 55.02 2.84 62.32 ...\n",
      "  .. ..$ Strain het         : num [1:82] 33.3 20 27.3 33.3 15.2 ...\n",
      "  .. ..$ Coverage ECHO_Blank: num [1:82] 2.65 3.64 4.51 9.28 2.65 ...\n",
      "  .. ..$ TPM ECHO_Blank     : num [1:82] 4855 6684 8282 17024 4864 ...\n",
      "  .. ..$ Coverage ECHO_C    : num [1:82] 2.33 29.2 4.39 2.96 2.01 ...\n",
      "  .. ..$ TPM ECHO_C         : num [1:82] 5233 65421 9846 6629 4509 ...\n",
      "  .. ..$ Coverage ECHO_F    : num [1:82] 5.65 5.73 3.86 1.65 2.55 ...\n",
      "  .. ..$ TPM ECHO_F         : num [1:82] 14183 14382 9694 4144 6388 ...\n",
      "  ..$ tpm  : num [1:82, 1:3] 4855 6684 8282 17024 4864 ...\n",
      "  .. ..- attr(*, \"dimnames\")=List of 2\n",
      "  .. .. ..$ : chr [1:82] \"metabat2.134.fa.contigs\" \"metabat2.53.fa.contigs\" \"maxbin.073.fasta.contigs\" \"metabat2.64.fa.contigs\" ...\n",
      "  .. .. ..$ : chr [1:3] \"ECHO_Blank\" \"ECHO_C\" \"ECHO_F\"\n",
      "  ..$ tax  : chr [1:82, 1:7] \"Bacteria\" \"Bacteria\" \"Bacteria\" \"Bacteria\" ...\n",
      "  .. ..- attr(*, \"dimnames\")=List of 2\n",
      "  .. .. ..$ : chr [1:82] \"maxbin.004.fasta.contigs\" \"maxbin.006.fasta.contigs\" \"maxbin.010.fasta.contigs\" \"maxbin.019.fasta.contigs\" ...\n",
      "  .. .. ..$ : chr [1:7] \"superkingdom\" \"phylum\" \"class\" \"order\" ...\n",
      " $ taxa       :List of 7\n",
      "  ..$ superkingdom:List of 2\n",
      "  .. ..$ abund  : int [1:7, 1:3] 199836 43313643 43874 80871 5250999 33795808 39941 166501 45521937 50822 ...\n",
      "  .. .. ..- attr(*, \"dimnames\")=List of 2\n",
      "  .. .. .. ..$ : chr [1:7] \"Archaea\" \"Bacteria\" \"Eukaryota\" \"No CDS\" ...\n",
      "  .. .. .. ..$ : chr [1:3] \"ECHO_Blank\" \"ECHO_C\" \"ECHO_F\"\n",
      "  .. ..$ percent: num [1:7, 1:3] 0.2416 52.3586 0.053 0.0978 6.3475 ...\n",
      "  .. .. ..- attr(*, \"dimnames\")=List of 2\n",
      "  .. .. .. ..$ : chr [1:7] \"Archaea\" \"Bacteria\" \"Eukaryota\" \"No CDS\" ...\n",
      "  .. .. .. ..$ : chr [1:3] \"ECHO_Blank\" \"ECHO_C\" \"ECHO_F\"\n",
      "  ..$ phylum      :List of 2\n",
      "  .. ..$ abund  : int [1:242, 1:3] 78 2356 369 28 4 18 6 217 99 3 ...\n",
      "  .. .. ..- attr(*, \"dimnames\")=List of 2\n",
      "  .. .. .. ..$ : chr [1:242] \"Candidatus Aenigmarchaeota\" \"Candidatus Altiarchaeota\" \"Candidatus Bathyarchaeota\" \"Candidatus Diapherotrites\" ...\n",
      "  .. .. .. ..$ : chr [1:3] \"ECHO_Blank\" \"ECHO_C\" \"ECHO_F\"\n",
      "  .. ..$ percent: num [1:242, 1:3] 9.43e-05 2.85e-03 4.46e-04 3.38e-05 4.84e-06 ...\n",
      "  .. .. ..- attr(*, \"dimnames\")=List of 2\n",
      "  .. .. .. ..$ : chr [1:242] \"Candidatus Aenigmarchaeota\" \"Candidatus Altiarchaeota\" \"Candidatus Bathyarchaeota\" \"Candidatus Diapherotrites\" ...\n",
      "  .. .. .. ..$ : chr [1:3] \"ECHO_Blank\" \"ECHO_C\" \"ECHO_F\"\n",
      "  ..$ class       :List of 2\n",
      "  .. ..$ abund  : int [1:542, 1:3] 78 1526 830 16 353 28 4 18 6 217 ...\n",
      "  .. .. ..- attr(*, \"dimnames\")=List of 2\n",
      "  .. .. .. ..$ : chr [1:542] \"Unclassified Candidatus Aenigmarchaeota\" \"Candidatus Altiarchaeales (no class in NCBI)\" \"Unclassified Candidatus Altiarchaeota\" \"Candidatus Bathyarchaeota archaeon (no class in NCBI)\" ...\n",
      "  .. .. .. ..$ : chr [1:3] \"ECHO_Blank\" \"ECHO_C\" \"ECHO_F\"\n",
      "  .. ..$ percent: num [1:542, 1:3] 9.43e-05 1.84e-03 1.00e-03 1.93e-05 4.27e-04 ...\n",
      "  .. .. ..- attr(*, \"dimnames\")=List of 2\n",
      "  .. .. .. ..$ : chr [1:542] \"Unclassified Candidatus Aenigmarchaeota\" \"Candidatus Altiarchaeales (no class in NCBI)\" \"Unclassified Candidatus Altiarchaeota\" \"Candidatus Bathyarchaeota archaeon (no class in NCBI)\" ...\n",
      "  .. .. .. ..$ : chr [1:3] \"ECHO_Blank\" \"ECHO_C\" \"ECHO_F\"\n",
      "  ..$ order       :List of 2\n",
      "  .. ..$ abund  : int [1:978, 1:3] 78 1526 830 16 353 28 4 18 6 217 ...\n",
      "  .. .. ..- attr(*, \"dimnames\")=List of 2\n",
      "  .. .. .. ..$ : chr [1:978] \"Unclassified Candidatus Aenigmarchaeota\" \"Candidatus Altiarchaeales\" \"Unclassified Candidatus Altiarchaeota\" \"Candidatus Bathyarchaeota archaeon (no order in NCBI)\" ...\n",
      "  .. .. .. ..$ : chr [1:3] \"ECHO_Blank\" \"ECHO_C\" \"ECHO_F\"\n",
      "  .. ..$ percent: num [1:978, 1:3] 9.43e-05 1.84e-03 1.00e-03 1.93e-05 4.27e-04 ...\n",
      "  .. .. ..- attr(*, \"dimnames\")=List of 2\n",
      "  .. .. .. ..$ : chr [1:978] \"Unclassified Candidatus Aenigmarchaeota\" \"Candidatus Altiarchaeales\" \"Unclassified Candidatus Altiarchaeota\" \"Candidatus Bathyarchaeota archaeon (no order in NCBI)\" ...\n",
      "  .. .. .. ..$ : chr [1:3] \"ECHO_Blank\" \"ECHO_C\" \"ECHO_F\"\n",
      "  ..$ family      :List of 2\n",
      "  .. ..$ abund  : int [1:1751, 1:3] 78 0 1526 830 16 353 28 4 18 6 ...\n",
      "  .. .. ..- attr(*, \"dimnames\")=List of 2\n",
      "  .. .. .. ..$ : chr [1:1751] \"Unclassified Candidatus Aenigmarchaeota\" \"Candidatus Altiarchaeum (no family in NCBI)\" \"Unclassified Candidatus Altiarchaeales\" \"Unclassified Candidatus Altiarchaeota\" ...\n",
      "  .. .. .. ..$ : chr [1:3] \"ECHO_Blank\" \"ECHO_C\" \"ECHO_F\"\n",
      "  .. ..$ percent: num [1:1751, 1:3] 9.43e-05 0.00 1.84e-03 1.00e-03 1.93e-05 ...\n",
      "  .. .. ..- attr(*, \"dimnames\")=List of 2\n",
      "  .. .. .. ..$ : chr [1:1751] \"Unclassified Candidatus Aenigmarchaeota\" \"Candidatus Altiarchaeum (no family in NCBI)\" \"Unclassified Candidatus Altiarchaeales\" \"Unclassified Candidatus Altiarchaeota\" ...\n",
      "  .. .. .. ..$ : chr [1:3] \"ECHO_Blank\" \"ECHO_C\" \"ECHO_F\"\n",
      "  ..$ genus       :List of 2\n",
      "  .. ..$ abund  : int [1:3688, 1:3] 78 0 1526 830 16 353 28 4 18 6 ...\n",
      "  .. .. ..- attr(*, \"dimnames\")=List of 2\n",
      "  .. .. .. ..$ : chr [1:3688] \"Unclassified Candidatus Aenigmarchaeota\" \"Candidatus Altiarchaeum\" \"Unclassified Candidatus Altiarchaeales\" \"Unclassified Candidatus Altiarchaeota\" ...\n",
      "  .. .. .. ..$ : chr [1:3] \"ECHO_Blank\" \"ECHO_C\" \"ECHO_F\"\n",
      "  .. ..$ percent: num [1:3688, 1:3] 9.43e-05 0.00 1.84e-03 1.00e-03 1.93e-05 ...\n",
      "  .. .. ..- attr(*, \"dimnames\")=List of 2\n",
      "  .. .. .. ..$ : chr [1:3688] \"Unclassified Candidatus Aenigmarchaeota\" \"Candidatus Altiarchaeum\" \"Unclassified Candidatus Altiarchaeales\" \"Unclassified Candidatus Altiarchaeota\" ...\n",
      "  .. .. .. ..$ : chr [1:3] \"ECHO_Blank\" \"ECHO_C\" \"ECHO_F\"\n",
      "  ..$ species     :List of 2\n",
      "  .. ..$ abund  : int [1:6252, 1:3] 78 0 1526 830 16 353 28 4 18 6 ...\n",
      "  .. .. ..- attr(*, \"dimnames\")=List of 2\n",
      "  .. .. .. ..$ : chr [1:6252] \"Unclassified Candidatus Aenigmarchaeota\" \"Candidatus Altiarchaeum sp.\" \"Unclassified Candidatus Altiarchaeales\" \"Unclassified Candidatus Altiarchaeota\" ...\n",
      "  .. .. .. ..$ : chr [1:3] \"ECHO_Blank\" \"ECHO_C\" \"ECHO_F\"\n",
      "  .. ..$ percent: num [1:6252, 1:3] 9.43e-05 0.00 1.84e-03 1.00e-03 1.93e-05 ...\n",
      "  .. .. ..- attr(*, \"dimnames\")=List of 2\n",
      "  .. .. .. ..$ : chr [1:6252] \"Unclassified Candidatus Aenigmarchaeota\" \"Candidatus Altiarchaeum sp.\" \"Unclassified Candidatus Altiarchaeales\" \"Unclassified Candidatus Altiarchaeota\" ...\n",
      "  .. .. .. ..$ : chr [1:3] \"ECHO_Blank\" \"ECHO_C\" \"ECHO_F\"\n",
      " $ functions  :List of 3\n",
      "  ..$ KEGG:List of 5\n",
      "  .. ..$ abund      : int [1:12479, 1:3] 19700 606 12461 3169 92 4 24 11932 65 22961 ...\n",
      "  .. .. ..- attr(*, \"dimnames\")=List of 2\n",
      "  .. .. .. ..$ : chr [1:12479] \"K00001\" \"K00002\" \"K00003\" \"K00004\" ...\n",
      "  .. .. .. ..$ : chr [1:3] \"ECHO_Blank\" \"ECHO_C\" \"ECHO_F\"\n",
      "  .. ..$ bases      : num [1:12479, 1:3] 2678362 81224 1714973 434493 12761 ...\n",
      "  .. .. ..- attr(*, \"dimnames\")=List of 2\n",
      "  .. .. .. ..$ : chr [1:12479] \"K00001\" \"K00002\" \"K00003\" \"K00004\" ...\n",
      "  .. .. .. ..$ : chr [1:3] \"ECHO_Blank\" \"ECHO_C\" \"ECHO_F\"\n",
      "  .. ..$ cov        : num [1:12478, 1:3] 4387.9 157.1 2424.7 712.4 18.8 ...\n",
      "  .. .. ..- attr(*, \"dimnames\")=List of 2\n",
      "  .. .. .. ..$ : chr [1:12478] \"K00001\" \"K00002\" \"K00003\" \"K00004\" ...\n",
      "  .. .. .. ..$ : chr [1:3] \"ECHO_Blank\" \"ECHO_C\" \"ECHO_F\"\n",
      "  .. ..$ tpm        : num [1:12479, 1:3] 223.45 7.84 127.38 33.65 1.05 ...\n",
      "  .. .. ..- attr(*, \"dimnames\")=List of 2\n",
      "  .. .. .. ..$ : chr [1:12479] \"K00001\" \"K00002\" \"K00003\" \"K00004\" ...\n",
      "  .. .. .. ..$ : chr [1:3] \"ECHO_Blank\" \"ECHO_C\" \"ECHO_F\"\n",
      "  .. ..$ copy_number: num [1:12478, 1:3] 1.3582 0.0486 0.7505 0.2205 0.0058 ...\n",
      "  .. .. ..- attr(*, \"dimnames\")=List of 2\n",
      "  .. .. .. ..$ : chr [1:12478] \"K00001\" \"K00002\" \"K00003\" \"K00004\" ...\n",
      "  .. .. .. ..$ : chr [1:3] \"ECHO_Blank\" \"ECHO_C\" \"ECHO_F\"\n",
      "  ..$ COG :List of 5\n",
      "  .. ..$ abund      : int [1:55336, 1:3] 26367 10925 7889 844 10075 40198 8874 27320 9347 12568 ...\n",
      "  .. .. ..- attr(*, \"dimnames\")=List of 2\n",
      "  .. .. .. ..$ : chr [1:55336] \"COG0001\" \"COG0002\" \"COG0003\" \"COG0004\" ...\n",
      "  .. .. .. ..$ : chr [1:3] \"ECHO_Blank\" \"ECHO_C\" \"ECHO_F\"\n",
      "  .. ..$ bases      : num [1:55336, 1:3] 3640701 1447368 1081385 116234 1313783 ...\n",
      "  .. .. ..- attr(*, \"dimnames\")=List of 2\n",
      "  .. .. .. ..$ : chr [1:55336] \"COG0001\" \"COG0002\" \"COG0003\" \"COG0004\" ...\n",
      "  .. .. .. ..$ : chr [1:3] \"ECHO_Blank\" \"ECHO_C\" \"ECHO_F\"\n",
      "  .. ..$ cov        : num [1:55335, 1:3] 5506 2411 1586 224 2461 ...\n",
      "  .. .. ..- attr(*, \"dimnames\")=List of 2\n",
      "  .. .. .. ..$ : chr [1:55335] \"COG0001\" \"COG0002\" \"COG0003\" \"COG0004\" ...\n",
      "  .. .. .. ..$ : chr [1:3] \"ECHO_Blank\" \"ECHO_C\" \"ECHO_F\"\n",
      "  .. ..$ tpm        : num [1:55336, 1:3] 284.85 121.03 84.24 9.75 117.49 ...\n",
      "  .. .. ..- attr(*, \"dimnames\")=List of 2\n",
      "  .. .. .. ..$ : chr [1:55336] \"COG0001\" \"COG0002\" \"COG0003\" \"COG0004\" ...\n",
      "  .. .. .. ..$ : chr [1:3] \"ECHO_Blank\" \"ECHO_C\" \"ECHO_F\"\n",
      "  .. ..$ copy_number: num [1:55335, 1:3] 1.7043 0.7464 0.4909 0.0693 0.7618 ...\n",
      "  .. .. ..- attr(*, \"dimnames\")=List of 2\n",
      "  .. .. .. ..$ : chr [1:55335] \"COG0001\" \"COG0002\" \"COG0003\" \"COG0004\" ...\n",
      "  .. .. .. ..$ : chr [1:3] \"ECHO_Blank\" \"ECHO_C\" \"ECHO_F\"\n",
      "  ..$ PFAM:List of 5\n",
      "  .. ..$ abund      : int [1:8080, 1:3] 7 59326 448433 24588 43 55268 8581 25428 88 8 ...\n",
      "  .. .. ..- attr(*, \"dimnames\")=List of 2\n",
      "  .. .. .. ..$ : chr [1:8080] \"PF00003 [7 transmembrane sweet-taste receptor of 3 GCPR]\" \"PF00004 [ATPase family associated with various cellular activities (AAA)]\" \"PF00005 [ABC transporter]\" \"PF00006 [ATP synthase alpha/beta family, nucleotide-binding domain]\" ...\n",
      "  .. .. .. ..$ : chr [1:3] \"ECHO_Blank\" \"ECHO_C\" \"ECHO_F\"\n",
      "  .. ..$ bases      : num [1:8080, 1:3] 921 8518663 60753090 3528247 6197 ...\n",
      "  .. .. ..- attr(*, \"dimnames\")=List of 2\n",
      "  .. .. .. ..$ : chr [1:8080] \"PF00003 [7 transmembrane sweet-taste receptor of 3 GCPR]\" \"PF00004 [ATPase family associated with various cellular activities (AAA)]\" \"PF00005 [ABC transporter]\" \"PF00006 [ATP synthase alpha/beta family, nucleotide-binding domain]\" ...\n",
      "  .. .. .. ..$ : chr [1:3] \"ECHO_Blank\" \"ECHO_C\" \"ECHO_F\"\n",
      "  .. ..$ cov        : num [1:8079, 1:3] 1.94 7635.04 77264.03 4313.49 16.04 ...\n",
      "  .. .. ..- attr(*, \"dimnames\")=List of 2\n",
      "  .. .. .. ..$ : chr [1:8079] \"PF00003 [7 transmembrane sweet-taste receptor of 3 GCPR]\" \"PF00004 [ATPase family associated with various cellular activities (AAA)]\" \"PF00005 [ABC transporter]\" \"PF00006 [ATP synthase alpha/beta family, nucleotide-binding domain]\" ...\n",
      "  .. .. .. ..$ : chr [1:3] \"ECHO_Blank\" \"ECHO_C\" \"ECHO_F\"\n",
      "  .. ..$ tpm        : num [1:8080, 1:3] 7.75e-02 5.19e+02 3.65e+03 2.41e+02 8.30e-01 ...\n",
      "  .. .. ..- attr(*, \"dimnames\")=List of 2\n",
      "  .. .. .. ..$ : chr [1:8080] \"PF00003 [7 transmembrane sweet-taste receptor of 3 GCPR]\" \"PF00004 [ATPase family associated with various cellular activities (AAA)]\" \"PF00005 [ABC transporter]\" \"PF00006 [ATP synthase alpha/beta family, nucleotide-binding domain]\" ...\n",
      "  .. .. .. ..$ : chr [1:3] \"ECHO_Blank\" \"ECHO_C\" \"ECHO_F\"\n",
      "  .. ..$ copy_number: num [1:8079, 1:3] 6.01e-04 2.36 2.39e+01 1.34 4.97e-03 ...\n",
      "  .. .. ..- attr(*, \"dimnames\")=List of 2\n",
      "  .. .. .. ..$ : chr [1:8079] \"PF00003 [7 transmembrane sweet-taste receptor of 3 GCPR]\" \"PF00004 [ATPase family associated with various cellular activities (AAA)]\" \"PF00005 [ABC transporter]\" \"PF00006 [ATP synthase alpha/beta family, nucleotide-binding domain]\" ...\n",
      "  .. .. .. ..$ : chr [1:3] \"ECHO_Blank\" \"ECHO_C\" \"ECHO_F\"\n",
      " $ total_reads: Named num [1:3] 82724972 84809672 84726428\n",
      "  ..- attr(*, \"names\")= chr [1:3] \"ECHO_Blank\" \"ECHO_C\" \"ECHO_F\"\n",
      " - attr(*, \"class\")= chr \"SQM\"\n"
     ]
    }
   ],
   "source": [
    "str(oil_spades)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "mineral-mineral",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in dyn.load(file, DLLpath = DLLpath, ...): unable to load shared object '/home/hwlee/anaconda3/envs/SqueezeMeta/lib/R/library/stringi/libs/stringi.so':\n  libicui18n.so.58: cannot open shared object file: No such file or directory\n",
     "output_type": "error",
     "traceback": [
      "Error in dyn.load(file, DLLpath = DLLpath, ...): unable to load shared object '/home/hwlee/anaconda3/envs/SqueezeMeta/lib/R/library/stringi/libs/stringi.so':\n  libicui18n.so.58: cannot open shared object file: No such file or directory\nTraceback:\n",
      "1. plotTaxonomy(oil_spades, rank = \"species\", count = \"percent\")",
      "2. plotBars(data, label_y = nice_label, color = color, label_fill = nice_rank, \n .     base_size = base_size, max_scale_value = max_scale_value)",
      "3. reshape2::melt",
      "4. getExportedValue(pkg, name)",
      "5. asNamespace(ns)",
      "6. getNamespace(ns)",
      "7. loadNamespace(name)",
      "8. namespaceImport(ns, loadNamespace(i, c(lib.loc, .libPaths()), \n .     versionCheck = vI[[i]]), from = package)",
      "9. loadNamespace(i, c(lib.loc, .libPaths()), versionCheck = vI[[i]])",
      "10. namespaceImport(ns, loadNamespace(i, c(lib.loc, .libPaths()), \n  .     versionCheck = vI[[i]]), from = package)",
      "11. loadNamespace(i, c(lib.loc, .libPaths()), versionCheck = vI[[i]])",
      "12. library.dynam(lib, package, package.lib)",
      "13. dyn.load(file, DLLpath = DLLpath, ...)"
     ]
    }
   ],
   "source": [
    "plotTaxonomy(oil_spades, rank='species', count='percent')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cellular-postage",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "3.6.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
